crates/telio-wg/src/wg.rs

```diff
@@ -292,7 +296,26 @@ impl WireGuard for DynamicWg {
         loop {
             if let Some(port) = self.get_interface().await?.listen_port {
                 if port > 0 {
-                    telio_log_trace!("Wireguard using port:{:?}", &port.to_string());
+                    telio_log_debug!("Wireguard using port:{:?}", &port.to_string());
+                    break Ok(port);
+                }
+            }
+            let elapsed_time = start.elapsed()?;
+            if elapsed_time > d {
+                break telio_err_with_log!(Error::PortAssignmentTimeoutError);
+            }
+
+            sleep(Duration::from_millis(10)).await;
```

Comment:
Wow, that's quite hacky

Reply:
Yeh, sadly there is no signal from wireguard that listen port was set :cry: